File: 1968-array-with-elements-not-equal-to-average-of-neighbors/1968-array-with-elements-not-equal-to-average-of-neighbors.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int n = nums.length;
        PriorityQueue<Integer> minHeap = new PriorityQueue<>();
        for(int num : nums) minHeap.add(-1 * num);
        for(int i = 1; i < n; i += 2) nums[i] = -1 * minHeap.poll();
        for(int i = 0; i < n; i += 2) nums[i] = -1 * minHeap.poll();
        return nums;
    }
}

File: 1968-array-with-elements-not-equal-to-average-of-neighbors/1968-array-with-elements-not-equal-to-average-of-neighbors.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        n, heap = len(nums), []
        for num in nums: 
            heappush(heap, -1 * num)
        for i in range (1, n, 2):
            nums[i] = -1 * heap[0]
            heappop(heap)
        for i in range (0, n, 2):
            nums[i] = -1 * heap[0]
            heappop(heap)
        return nums